internal/commands/watch.go

```diff
@@ -0,0 +1,79 @@
+package commands
+
+import (
+	"context"
+	"time"
+
+	"github.com/authzed/zed/internal/client"
+	"github.com/authzed/zed/internal/console"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"github.com/spf13/cobra"
+)
+
+var (
+	watchObjectTypes []string
+	watchRevision    string
+	watchTimestamps  bool
+)
+
+func RegisterWatchCmd(rootCmd *cobra.Command) *cobra.Command {
+	rootCmd.AddCommand(watchCmd)
+
+	watchCmd.Flags().StringSliceVar(&watchObjectTypes, "object_types", nil, "optional object types to watch updates for")
+	watchCmd.Flags().StringVar(&watchRevision, "revision", "", "optional revision at which to start watching")
+	watchCmd.Flags().BoolVar(&watchTimestamps, "timestamp", false, "shows timestamp of incoming update events")
+	return watchCmd
+}
+
+var watchCmd = &cobra.Command{
+	Use:   "watch [object_types, ...] [start_cursor]",
+	Short: "watches the stream of relationship updates from the server",
+	Args:  cobra.RangeArgs(0, 2),
+	RunE:  watchCmdFunc,
+}
+
+func watchCmdFunc(cmd *cobra.Command, _ []string) error {
+	console.Errorf("starting watch stream over types %v and revision %v\n", watchObjectTypes, watchRevision)
+
+	cli, err := client.NewClient(cmd)
+	if err != nil {
+		return err
+	}
+
+	req := &v1.WatchRequest{
+		OptionalObjectTypes: watchObjectTypes,
+	}
+	if watchRevision != "" {
+		req.OptionalStartCursor = &v1.ZedToken{Token: watchRevision}
+	}
+
+	ctx, cancel := context.WithCancel(cmd.Context())
```

Comment:
don't we still need something like https://github.com/authzed/spicedb/blob/64e3c41567024931d4ae64482299c2ba90452544/pkg/cmd/signals.go#L16 ?

since watch is long running, the only way cancel will get called is if we tie it to the os signals, right?

Reply:
yeah, that or if we provide a `--deadline` flag. I can give that a go.